.github/actions/build-push-artifacts/action.yml

```diff
@@ -0,0 +1,63 @@
+name: 'Builds and Push Artifacts'
+description: 'Builds and pushes a Docker image, S3 env vars file, and dispatches a Twistlock workflow'
+inputs:
+  environment:
+    required: true
+    default: "dev"
+    type: string
+  ref:
+    description: 'Git reference (commit SHA, branch, or tag) to build'
+    required: true
+  aws-access-key-id:
+    description: 'AWS Access Key ID'
+    required: true
+  aws-secret-access-key:
+    description: 'AWS Secret Access Key'
+    required: true
+  role-to-assume:
+    description: 'AWS role to assume'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ inputs.ref }}
+
+    - name: Configure VAEC AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v4.0.2
```

Comment:
We don't usually specify beyond the major version. Does patch 2 gives us something special?